wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -638,9 +666,58 @@ class PlatformRepo(val walletApplication: WalletApplication) {
                     dashPayProfileDaoAsync.insert(profile!!)
                 }
             }
+            queueContactsUpdatedListeners()
             log.info("updating contacts and profiles took $watch")
         } catch (e: Exception) {
-            log.error("error updating contacts ${e.message}")
+            log.error(formatExceptionMessage("error updating contacts", e))
+        } finally {
+            updatingContacts.set(false)
+        }
+    }
+
+    // Create the Handler object (on the main thread by default)
+    val timerHandler = Handler()
+    var timerStarted = false
+
+    // Define the code block to be executed
+    private val executeUpdateContacts = object : Runnable {
+        override fun run() {
+            // Do something here on the main thread
+            log.info("Timer: Update contacts")
+            // Repeat this the same runnable code block again another 2 seconds
+            GlobalScope.launch {
+                updateContactRequests()
+            }
+            timerHandler.postDelayed(this, UPDATE_TIMER_DELAY)
+        }
+    }
+
+    // Start the initial runnable task by posting through the handler
+    fun startUpdateTimer() {
+        log.info("Starting timer for updating DashPay")
+        if (!timerStarted) {
+            timerHandler.post(executeUpdateContacts)
+            timerStarted = true
+        }
+    }
+
+    fun stopUpdateTimer() {
+        timerStarted = false
+        timerHandler.removeCallbacks(executeUpdateContacts)
+    }
+
+    fun addContactsUpdatedListener(listener: OnContactsUpdated) {
+        onContactsUpdatedListeners.add(listener)
+    }
+
+    fun removeContactsUpdatedListener(listener: OnContactsUpdated?) {
+        onContactsUpdatedListeners.remove(listener)
+    }
+
+    fun queueContactsUpdatedListeners() {
```

Comment:
It doesn't look like we're queuing calls on this method, so I would rename to `fireContactsUpdatedListeners` or something similar. Also, this function can be private.

Reply:
Fixed here:
https://github.com/dashevo/dash-wallet/pull/460/commits/929801ebf4d37deb29e65188d61588f5ddc7c7f1